lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -316,57 +341,44 @@ def add_test_targets
           # @note   The source files are grouped by Pod and in turn by subspec
           #         (recursively) in the resources group.
           #
-          # @return [void]
+          # @param  [Array<Sandbox::FileAccessor>] file_accessors
+          #         the file accessors list to generate resource bundles for.
           #
-          def add_resources_bundle_targets
-            target.file_accessors.each do |file_accessor|
-              file_accessor.resource_bundles.each do |bundle_name, paths|
+          # @return [Array<PBXNativeTarget] the resource bundle native targets created.
+          #
+          def add_resources_bundle_targets(file_accessors)
+            file_accessors.flat_map do |file_accessor|
+              file_accessor.resource_bundles.map do |bundle_name, paths|
                 label = target.resources_bundle_target_label(bundle_name)
-                bundle_target = project.new_resources_bundle(label, file_accessor.spec_consumer.platform_name)
-                bundle_target.product_reference.tap do |bundle_product|
+                resource_bundle_target = project.new_resources_bundle(label, file_accessor.spec_consumer.platform_name)
+                resource_bundle_target.product_reference.tap do |bundle_product|
                   bundle_file_name = "#{bundle_name}.bundle"
                   bundle_product.name = bundle_file_name
                 end
 
                 filter_resource_file_references(paths) do |resource_phase_refs, compile_phase_refs|
                   # Resource bundles are only meant to have resources, so install everything
                   # into the resources phase. See note in filter_resource_file_references.
-                  bundle_target.add_resources(resource_phase_refs + compile_phase_refs)
+                  resource_bundle_target.add_resources(resource_phase_refs + compile_phase_refs)
                 end
 
-                native_target = target.native_target_for_spec(file_accessor.spec_consumer.spec)
                 target.user_build_configurations.each do |bc_name, type|
-                  bundle_target.add_build_configuration(bc_name, type)
-                end
-                bundle_target.deployment_target = deployment_target
-
-                test_specification = file_accessor.spec.test_specification?
-
-                if test_specification
-                  target.test_resource_bundle_targets << bundle_target
-                else
-                  target.resource_bundle_targets << bundle_target
-                end
-
-                if target.should_build?
-                  native_target.add_dependency(bundle_target)
```

Comment:
this logic has moved into `pods_project_generator` for clarity. In the past we were creating resource bundle targets here during installation but we were also wiring the dependencies up. This made it hard to follow. This class is solely responsible for *creating* targets and configuring them, it should not do dependencies as it does not have enough information to do so.

This method has moved and a test has been added.